UserAgeCalculation

```diff
@@ -9,22 +9,52 @@ namespace UserAgeCalculation
             Console.Write("Enter your name: ");
             string userName = Console.ReadLine();
 
+            int currentYear = DateTime.Now.Year;
 
-            Console.Write("Enter your year of birth: ");
-            int yearOfBirth = Convert.ToInt32(Console.ReadLine());
+            int yearOfBirth;
 
-            int currentYear = DateTime.Now.Year;
-            int userAge = currentYear - yearOfBirth; 
+            while (true)
+            {
+                Console.Write("Enter your year of birth: ");
+                string yearOfBirthString = Console.ReadLine();
+
+                bool yearOfBirthBool = int.TryParse(yearOfBirthString, out yearOfBirth);
+
+                if (yearOfBirth > 0)
+                {
+                    if (yearOfBirth <= currentYear - 100)
+                    {
+                        Console.WriteLine($"Error, your year of birth cannot be less than {currentYear - 100} or greater than { currentYear }, please enter valid value");
+                        continue;
+                    }
+
+                    if (yearOfBirth >= currentYear)
+                    {
+                        Console.WriteLine($"Error, your year of birth cannot be less than {currentYear - 100} or greater than { currentYear }, please enter valid value");
+                        continue;
+                    }
+                }
+
+                else
+                {
```

Comment:
move this validation before if-s. I recommend to make sure that entered value is what your program really expect.